packages/workbox-google-analytics/src/index.js

```diff
@@ -55,84 +50,128 @@ import replayQueuedRequests from './lib/replay-queued-requests.js';
- *                    prior to replaying the hit. The function is invoked with
- *                    the original hit's URLSearchParams object as its only
- *                    argument. To abort the hit and prevent it from being
- *                    replayed, throw an error.
- * @memberof module:workbox-google-analytics
+ * Creates the requestWillDequeue callback to be used with the background
+ * sync queue plugin. The callback takes the failed request and adds the
+ * `qt` param based on the current time, as well as applies any other
+ * user-defined hit modifications.
+ * @param {Object} config See workbox.googleAnalytics.initialize.
+ * @return {Function} The requestWillDequeu callback function.
  */
-const initialize = (config) => {
-  config = config || {};
-
-  // Stores whether or not the previous /collect request failed.
-  let previousHitFailed = false;
-
-  self.addEventListener('fetch', (event) => {
-    const url = new URL(event.request.url);
-    const request = event.request;
-
-    if (url.hostname === constants.URL.HOST) {
-      if (url.pathname === constants.URL.COLLECT_PATH) {
-        // If this is a /collect request, then use a network-first strategy,
-        // falling back to queueing the request in IndexedDB.
-
-        // Make a clone of the request before we use it, in case we need
-        // to read the request body later on.
-        const clonedRequest = request.clone();
-
-        event.respondWith(
-          fetch(request).then((response) => {
-            if (previousHitFailed) {
-              replayQueuedRequests(config);
-            }
-            previousHitFailed = false;
-            return response;
-          }, (error) => {
-            logHelper.log('Enqueuing failed request...');
-            previousHitFailed = true;
-            return enqueueRequest(clonedRequest).then(() => Response.error());
-          })
-        );
-      } else if (url.pathname === constants.URL.ANALYTICS_JS_PATH) {
-        // If this is a request for the Google Analytics JavaScript library,
-        // use the network first, falling back to the previously cached copy.
-        event.respondWith(
-          caches.open(constants.CACHE_NAME).then((cache) => {
-            return fetch(request).then((response) => {
-              return cache.put(request, response.clone()).then(() => response);
-            }).catch((error) => {
-              logHelper.error(error);
-              return cache.match(request);
-            });
-          })
-        );
+const createRequestWillDequeueCallback = (config) => {
+  return (reqData) => {
+    const {request} = reqData;
+    const hitTime = reqData.metadata.creationTimestamp;
+    const queueTime = Date.now() - hitTime;
+
+    // Measurement protocol requests can set their payload parameters in either
+    // the URL query string (for GET requests) or the POST body.
+    const params = request.body ? new URLSearchParams(request.body) :
```

Comment:
This relies on [Chrome 61](https://www.chromestatus.com/feature/5737549880360960), which just made it into stable yesterday. I think that's fine, since it's also in Firefox 54+, but it's cutting it close 😄

Reply:
`request.body` here will be a string, so it should be OK. Is that what you're referencing?

Reply:
Okay, if it's a string then it's probably fine for older browsers too. Chrome 61 added in support for passing in an Object.